addon/utils/register-helpers.js

```diff
@@ -2,7 +2,6 @@ import Ember from 'ember';
 import WaypointComponent from 'ember-waypoints/components/waypoint';
 
 export default function() {
-  var helper = Ember.Handlebars.makeViewHelper(WaypointComponent);
-
-  Ember.Handlebars.registerHelper('waypoint', helper);
+  var helper = Ember.HTMLBars.makeViewHelper(WaypointComponent);
+  Ember.HTMLBars._registerHelper('waypoint', helper);
```

Comment:
The `Ember.Handlebars` namespace is actually public API and has been consistently supported from 1.0.0+.  We should still use it here.